code/AssetLib/M3D/m3d.h

```diff
@@ -976,18 +971,15 @@ M3D_INDEX _m3d_gettx(m3d_t *model, m3dread_t readfilecb, m3dfree_t freecb, char
     model->texture[i].d = NULL;
     if (buff) {
         if (buff[0] == 0x89 && buff[1] == 'P' && buff[2] == 'N' && buff[3] == 'G') {
-#ifdef STBI__PNG_TYPE
-            s.read_from_callbacks = 0;
-            s.img_buffer = s.img_buffer_original = (unsigned char *)buff;
-            s.img_buffer_end = s.img_buffer_original_end = (unsigned char *)buff + len;
             /* don't use model->texture[i].w directly, it's a uint16_t */
-            w = h = len = 0;
-            ri.bits_per_channel = 8;
-            model->texture[i].d = (uint8_t *)stbi__png_load(&s, (int *)&w, (int *)&h, (int *)&len, 0, &ri);
-            model->texture[i].w = (uint16_t)w;
-            model->texture[i].h = (uint16_t)h;
-            model->texture[i].f = (uint8_t)len;
-#endif
+            // stb_image in/out vars (set via pointer), native type "int"
+            int w = 0;
+            int h = 0;
+            int img_components = 0;
+            model->texture[i].d = stbi_load_from_memory(buff, len, &w, &h, &img_components, 0);
```

Comment:
What happens if the image loading fails? Will some other peace of code handle that?

Reply:
That's a good point.  Previously, since `STBI__PNG_TYPE` was not defined in the scope of the source file including this header, this code was completely inaccessible and never executed.  So there is a possibility of new bugs being exposed just for that reason in the code within the now-removed `#ifdef` guards.

But my testing indicates that most of the sample M3D models will load successfully with this PR (which simply replaces a private function in `stb_image` with an equivalent public one); without this PR, none of them will load at all -- so my hope would be to get this merged, then possibly raise a new issue drawing attention to the possibility that there may be some additional error checking required.

Reply:
(Also, this time the Sonar Cloud report seems to be failing the PR based on complaints in the old code, not the new code)

Reply:
@turol Please have another look when you have time to check the modification, thank you

Reply:
Looks reasonable

Reply:
Thank you, OK to approve?